operator/roles/kiali-deploy/defaults/main.yml

```
@@ -117,6 +117,9 @@ kiali_defaults:
       service: ""
       url: ""
 
+  for_display:
+    annotations:
```

I believe you are going to need to explicitly put an empty list here. This should be:
```suggestion
    annotations: []
```